Analysis/Tasks/PWGGA/myGammaTask3.cxx

```diff
@@ -0,0 +1,281 @@
+                return kFALSE;
+            }
+        }
+        // High Pt Pion rej
+        else
+        {
+            if( theTrack.tpcNSigmaEl() > fPIDnSigmaBelowElectronLine && theTrack.tpcNSigmaEl() < fPIDnSigmaAboveElectronLine && theTrack.tpcNSigmaPi()<fPIDnSigmaAbovePionLineHighPt)
+            {
+                registry.fill(HIST("IsPhotonSelected"), kPionRejHighMom);
+                return kFALSE;
+            }
+        }
+    }
+    return kTRUE;  
+}
+
+
+
+void ATask::init(InitContext const&)
+{
+    TAxis *lXaxis = registry.get<TH1>(HIST("IsPhotonSelected"))->GetXaxis();
+    for (size_t i=0; i< fPhotCutsLabels.size(); ++i)
+    {
+        lXaxis->SetBinLabel(i+1, fPhotCutsLabels[i]);
+    }
+}
+  
+void ATask::process(aod::Collision const& collision, 
+                    aod::V0Datas const& v0s, 
+                    tracksAndTPCInfo const& tracks,
+                    aod::McParticles const& mcParticles)
+{
+    for (auto& v0 : v0s) {
+        ++i;
+        float lPhiV0Rec = static_cast<float>(M_PI) + std::atan2(-v0.py(), -v0.px());
+        
+        // fill some QA histograms before any cuts
+        registry.fill(HIST("beforeCuts/hPtRec_before"), v0.pt());
+        registry.fill(HIST("beforeCuts/hEtaRec_before"), v0.eta());
+        registry.fill(HIST("beforeCuts/hPhiRec_before"), lPhiV0Rec);
+        registry.fill(HIST("beforeCuts/hConvPointR_before"), v0.v0radius());
+
+        registry.fill(HIST("IsPhotonSelected"), kPhotonIn);
+        
+        auto trackPos = v0.template posTrack_as<tracksAndTPCInfo>(); //positive daughter
+        auto trackNeg = v0.template negTrack_as<tracksAndTPCInfo>(); //positive daughter
+        
+        // single track eta cut
+        if (TMath::Abs(trackPos.eta()) > fEtaCut || TMath::Abs(trackNeg.eta()) > fEtaCut)
+        {
+            registry.fill(HIST("IsPhotonSelected"), kTrackEta);
+            continue;
+        }
+        
+        // single track pt cut    
+        if ( (TMath::Power(v0.pxpos(),2) + TMath::Power(v0.pypos(),2) < fSinglePtSq) ||
+             (TMath::Power(v0.pxneg(),2) + TMath::Power(v0.pyneg(),2) < fSinglePtSq)
+           )
+        {
+            registry.fill(HIST("IsPhotonSelected"), kTrackPt);
+            continue;
+        } 
+        
+        if (!(selectionPIDTPC_track(trackPos) && selectionPIDTPC_track(trackNeg)))
+        {
+            continue;
+        }
+        
+        if (v0.v0radius() < fMinR || v0.v0radius() > fMaxR)
+        {
+                registry.fill(HIST("IsPhotonSelected"), kV0Radius);
+                continue;
+        }
+        
+        Double_t lAlpha = -1.;
+        Double_t lQt = -1.;
+        computeArmenteros(v0, lAlpha, lQt);
+        //todo: check if values of lAlpha and Qt need to be checked
+        if (!ArmenterosQtCut(lAlpha, lQt, v0.pt()))
+        {
+            registry.fill(HIST("IsPhotonSelected"), kArmenteros);
+            continue;
+        }
+        
+        Double_t lPsiPair = computePsiPair(v0);
```

Comment:
I suggest that this quantity is also added to the V0Datas table as a dynamic or expression column. However, at the moment this table is produced by a task in LF. So you may want to discuss in the group and possibly also with David Chinellato if you want to introduce a dependency from PWG-LF. One solution could be to promote the lambdakzerobuilder task from LF to become a global task, for more PWGs to use.

Reply:
okay thanks, I will discuss with them

Reply:
Hi @iarsene,
I have a question regarding your comment: You suggest that the PsiPair quantity is added to the V0Datas table. This is one thing.
Another question is whether we want this task to have a dependency on a task (lambdakzerobuilder) in PWG-LF as it is the case now or if lambdakzerobuilder should become a global task. But this question is irrespective from the first issue with PsiPair, right?

Reply:
Hi, 
Regarding psiPair, I think this is something that is mostly up to you. My comment was in the context in which your task is dependent anyway on the LF task producing the V0Datas. So if you would like to continue in this direction, then it makes sense to move also the calculation of psiPair in the strangeness data model, since many of the quantities you need are already there.
Regarding the lambdakzerobuilder question (yes it is independent on the psiPair), i think this is a strategy decision you would need to take within GA at some point. And if you would like to use the lambdakzerobuilder as a V0Datas producer, then indeed i think this task could be promoted (or a new common one be made). However, this would depend also on David, since this task is part of his strangeness analysis and this would require some changes from his side as well.